### PR TITLE
Updated Guard article for the GuardAuthenticatorInterface deprecation

### LINKED ISSUE #8485: [Security][Guard] Deprecated GuardAuthenticatorInterface
As of 3.4

- `GuardAuthenticatorInterface` is deprecated and will be removed in 4.0, references should be updated to use `AuthenticatorInterface` instead (starting from 3.4);
- `AuthenticatorInterface` introduces a `supports()` method used to skip authenticators which are not  able to process authentication from the given request, any authenticator must implement it
- `GuardAuthenticatorInterface::getCredentials()` returning `null` causes the authenticator to be skipped. This behavior has been removed in the new `AuthenticatorInterface`, returning `null` from `AuthenticatorInterface::getCredentials()` throws an `\UnexpectedValueException` exception (`false` should be returned from `supports()` instead)

Code PR: https://github.com/symfony/symfony/pull/16835

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #8485.

### CODE REVIEW

File: security/guard_authentication.rst [hunk 1]
```diff
@@ -352,19 +365,27 @@ The Guard Authenticator Methods
...
 
+**supports(Request $request)**
+    This will be called on *every* request and your job is to decide if the
+    authenticator should be used for this request (return ``true``) or if it
+    should be skipped (return ``false``).
+
+.. versionadded:: 3.4
+    The ``supports()`` method was introduced in Symfony 3.4. In previous Symfony
+    versions, the authenticator could be skipped returning ``null`` in the
+    ``getCredentials()`` method.
```
Comment (xabbuh) [inline security/guard_authentication.rst]: These lines should be indented to be part of the definition item.

**Review by xabbuh — APPROVED** [review]

File: security/guard_authentication.rst [hunk 2]
```diff
@@ -352,19 +365,27 @@ The Guard Authenticator Methods
...
+        The ``supports()`` method was introduced in Symfony 3.4. In previous Symfony
+        versions, the authenticator could be skipped returning ``null`` in the
+        ``getCredentials()`` method.
+
 **getCredentials(Request $request)**
     This will be called on *every* request and your job is to read the token (or
     whatever your "authentication" information is) from the request and return it.
-    If you return ``null``, the rest of the authentication process is skipped. Otherwise,
-    ``getUser()`` will be called and the return value is passed as the first argument.
+    These credentials are later passed as the first argument ot ``getUser()``.
```
Comment (chalasr) [inline security/guard_authentication.rst]: typo `ot` => `of` :)

**Review by chalasr — APPROVED** [review]
👍  With minor comment

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**